query/src/sessions/context.rs

```diff
@@ -179,7 +181,14 @@ impl DatabendQueryContext {
     }
 
     pub fn set_current_database(&self, new_database_name: String) -> Result<()> {
-        match self.get_catalog().get_database(new_database_name.as_str()) {
+        let rt = self.shared.try_get_runtime()?;
+        let cata = self.get_catalog();
+        let db_name = new_database_name.clone();
+
+        let res = (async move { cata.get_database(&db_name).await })
+            .wait_in(&rt, Some(Duration::from_millis(5000)))?;
```

Comment:
I think `5000` should be a constant or feed it in the config file.

Reply:
Guess `get_database` should async soon :thinking: